lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+        systemCompilers.append(str(compilers))
+
+    for compiler in systemCompilers:
+        if compiler in newMaster:
+            newList.append(compiler)
+    return newList
+
+
+'''
+fucntion Name: removeTests
+parameters:
+    testList  : list
+    exclusion : str
+returns:
+    list
+
+Receieves a list of all tests and an excusion.
+Depending on the type of exclusion it will remove tests matching.
+Returns the list with the tests matching the excusion removed.
+
+exclusions can be in this form:
+pkg%compiler
+pkg@version
+compiler@version
+pkg@version%compiler@version
+pkg@version%compiler
+pkg%compiler@version
+pkg
+compiler
+'''
+def removeTests(testList, exclusion):
+    pkg = ""
+    compiler=""
+    teststoRemove = []
+    if "%" in exclusion:
+        pkg = exclusion.split('%')[0]
+        compiler=exclusion.split('%')[1]
+        if '@' in pkg and '@' in compiler:#pkg@version%compiler@version
+            if exclusion in testList:
+                teststoRemove.append(exclusion)
+        elif '@' in pkg or '@' in compiler:#pkg@version%compiler or pkg%compiler@version
+            for test in testList:
+                if pkg in test and compiler in test:
+                    teststoRemove.append(test)
+        else:#pkg%compiler
+            for test in testList:
+                if pkg in test and compiler in test:
+                    teststoRemove.append(test)
+    elif '@' in exclusion: #pkg@version or compiler@version
+        for test in testList:
+            if exclusion in test:
+                teststoRemove.append(test)
+    else:#pkg or compiler
+        remove=0
+        for test in testList:
+            if exclusion in str(test):
+                teststoRemove.append(test)
+    for test in teststoRemove: #remove tests from main test list
+        testList.remove(test)
+    return testList
+
+def testsuite(parser, args):
+    if not args.yamlFile:
+        tty.die("spack testsuite requires a yaml file as argument.")
+    if args.complete:
+        args.log_format='cdash-complete' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-complete'
+    else:
+        args.log_format='cdash-simple' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-simple'
+    print args.complete
+    enabledTests = []
+    compilers = []
+    packages = []
+    tests = []
+    testPackages = []
+    versions = []
+    packageVersion = []
+    compilerVersion = []
+    path = ""
+    dashboards = []
+    #designed to use a single file and modify the enabled tests, thus requiring a single file modification.
+    for files in args.yamlFile: #read yaml files which contains description of tests
+        with open(files, 'r') as stream:
+            try:
+                yamlFile= yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+            enabledTests= yamlFile['enable']
```

Comment:
The YAML file probably needs more validation, so that users see more than a `KeyError` when things go wrong (i.e., if they give you a file without `enable` in it).  We do that with `jsonschema` (note that `json` and `YAML` have basically the same data model, so this works).  See the schemata in [`lib/spack/spack/schema`](https://github.com/LLNL/spack/tree/develop/lib/spack/spack/schema), or the [corresponding docs](https://spack.readthedocs.io/en/latest/spack.schema.html).  Can you write one of those for your format and validate it when the YAML file gets read in?

You can look at [`spack.config.validate_section`](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/config.py#L139) for an example of how to validate against a `jsonschema`.  You can probably reuse/refactor [`ConfigFormatError`](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/config.py#L477) to make the schema validation print nice error messages with line numbers from the config file -- this leverages that `MarkedYAMLError` I mentioned up above.